### DISCUSSION TITLE
GC710 GC610 configuration conflict?

### DISCUSSION THREAD

**Discussion by wlmqljj** [OpenTabletDriver/OpenTabletDriver#2702] [Q&A] [answered]
I have a tablet Huion GC 710 runing on
Manjaro Kernel 6.1 AMD64 KDE Plasma Wayland/X11
Install via AUR

Shell lsusb:
Wireless dongle mode:
>Bus 001 Device 020: ID 256c:006d HUION Huion Tablet

Cable mode:
>Bus 001 Device 021: ID 256c:006e HUION Huion Tablet

I need to set `"ReportParser": "OpenTabletDriver.Configurations.Parsers.Huion.GianoReportParser"` to make wireless dongle(256c:006d) to work.
BTW, GC710 share same DeviceStrings with GC610. Because it is a GC610 with wireless module and battery soldered on.

My custom conf path: `/usr/share/OpenTabletDriver/Configurations/Huion/GC710.json`

Working badly when 
```json
"ButtonCount": 2
```
>1. Pen buttons not showing up. ( See attached pics )
>2. Auxiliary setting not work.

Working fine when
```json
"Buttons": {"ButtonCount": 2}
```
>Without problem above.

**Good Working Configuration:**
```json
 {
  "Name": "Huion GC710",
  "Specifications": {
    "Digitizer": {
      "Width": 254,
      "Height": 158.75,
      "MaxX": 50800,
      "MaxY": 31750
    },
    "Pen": {
      "MaxPressure": 8191,
      "Buttons": {
        "ButtonCount": 2
      }
    },
    "AuxiliaryButtons": {
      "ButtonCount": 6
    },
    "MouseButtons": null,
    "Touch": null
  },
  "DigitizerIdentifiers": [
    {
      "VendorID": 9580,
      "ProductID": 110,
      "InputReportLength": 12,
      "OutputReportLength": null,
      "ReportParser": "OpenTabletDriver.Configurations.Parsers.UCLogic.UCLogicReportParser",
      "FeatureInitReport": null,
      "OutputInitReport": null,
      "DeviceStrings": {
        "201": "HUION_T166_\\d{6}$"
      },
      "InitializationStrings": [
        200
      ]
    },
    {
      "VendorID": 9580,
      "ProductID": 109,
      "InputReportLength": 12,
      "OutputReportLength": null,
      "ReportParser": "OpenTabletDriver.Configurations.Parsers.Huion.GianoReportParser",
      "FeatureInitReport": null,
      "OutputInitReport": null,
      "DeviceStrings": {
        "201": "HUION_T166_\\d{6}$"
      },
      "InitializationStrings": [
        200
      ]
    }
  ],
  "AuxiliaryDeviceIdentifiers": [],
  "Attributes": {
    "libinputoverride": "1"
  }
}
```
**Bad Working Configuration:**
```json
{
  "Name": "Huion GC710",
  "Specifications": {
    "Digitizer": {
      "Width": 254,
      "Height": 158.75,
      "MaxX": 50800,
      "MaxY": 31750
    },
    "Pen": {
      "MaxPressure": 8191,
      "ButtonCount": 2
    },
    "AuxiliaryButtons": {
      "ButtonCount": 6
    },
    "MouseButtons": null,
    "Touch": null
  },
  "DigitizerIdentifiers": [
    {
      "VendorID": 9580,
      "ProductID": 110,
      "InputReportLength": 12,
      "OutputReportLength": null,
      "ReportParser": "OpenTabletDriver.Configurations.Parsers.UCLogic.UCLogicReportParser",
      "FeatureInitReport": null,
      "OutputInitReport": null,
      "DeviceStrings": {
        "201": "HUION_T166_\\d{6}$"
      },
      "InitializationStrings": [
        200
      ]
    },
    {
      "VendorID": 9580,
      "ProductID": 109,
      "InputReportLength": 12,
      "OutputReportLength": null,
      "ReportParser": "OpenTabletDriver.Configurations.Parsers.Huion.GianoReportParser",
      "FeatureInitReport": null,
      "OutputInitReport": null,
      "DeviceStrings": {
        "201": "HUION_T166_\\d{6}$"
      },
      "InitializationStrings": [
        200
      ]
    }
  ],
  "AuxiliaryDeviceIdentifiers": [],
  "Attributes": {
    "libinputoverride": "1"
  }
}
```
[Console_Debug.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/11239093/Console_Debug.txt)
[GOODLOG.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/11239047/GOODLOG.txt)
[BADLOG.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/11239048/BADLOG.txt)
![BAD](https://user-images.githubusercontent.com/16490196/232207113-6dc149b9-0524-4573-8a10-1e5c21b55096.png)
![GOOD](https://user-images.githubusercontent.com/16490196/232207117-65ac2a8e-983a-4d97-b9cc-12c3e9e1cd46.png)

**Comment by jamesbt365** [comment, marked as answer]
If the tablet has the same string, I'm assuming the internal specifications are also the same, if that is true you probably don't need the second configuration file and can just keep it all under one file.

> 1. Pen buttons not showing up. ( See attached pics )

The configuration format changed in master, it is not the same as 0.6.0.4 so if you're using the wrong format this is expected.
